tests/templates/test_subroutines.py

```diff
@@ -369,21 +369,22 @@ def test_single_ex_unitary_operations(self, ph, ref_gates):
         with qml.utils.OperationRecorder() as rec:
             SingleExcitationUnitary(weight, wires=ph)
 
-        idx = ref_gates[0][0]
+        assert len(rec.queue) == sqg + cnots            
 
-        exp_gate = ref_gates[0][1]
-        res_gate = rec.queue[idx]
+        for gate in ref_gates:
+            idx = gate[0]
 
-        exp_wires = ref_gates[0][2]
-        res_wires = rec.queue[idx]._wires
+            exp_gate = gate[1]
+            res_gate = rec.queue[idx]
```

Comment:
Do you need to define idx? Maybe just
`res_gate = rec.queue[gate[0]]`

Or even in a more compact form just use a list comprehension instead of the loop here.

Reply:
Thanks @soranjh.
I would prefer to keep the index ``idx``, otherwise I would need to add ``gate[0]`` multiple times to point to the elements of ``rec.queue`` below. Probably make things more compact but it degrades the readability of the unit test. These tests are very  tricky, I would keep them as clear as possible for us and others should we want to revisit them in the future.